pkg/kv/kvserver/replica_command.go

```diff
@@ -400,10 +406,8 @@ func (r *Replica) adminSplitWithDescriptor(
 		// range descriptors are picked outside the transaction. Return
 		// ConditionFailedError in the error detail so that the command can be
 		// retried.
-		if msg, ok := maybeDescriptorChangedError(desc, err); ok {
-			// NB: we have to wrap the existing error here as consumers of this
-			// code look at the root cause to sniff out the changed descriptor.
-			err = &benignError{errors.Wrap(err, msg)}
+		if dErr := maybeDescriptorChangedError(desc, true /*wrap*/, err); dErr != nil {
```

Comment:
[nit] Here and below, I have a slight preference for pulling out the wrapping logic out of maybeDescriptorChangedError and simply wrapping at the callsites, as before. I do like the return type being changed to an `error`, we should keep that.

Reply:
Maybe even keeping the comment block as to why we're wrapping vs. not is a good idea.